classes/notification/PKPNotificationOperationManager.inc.php

```diff
@@ -392,6 +412,69 @@ protected function sendNotificationEmail($request, $notification, ?int $contextI
 			}
 		}
 	}
+
+	/**
+	 * Creates and returns a unique string for the given notification, that will be encoded and validated against.
+	 * @param $notification Notification
+	 * @return string
+	 */
+	function createUnsubscribeUniqueKey($notification) {
+		$uniqueKey = 'unsubscribe' . '-' . $notification->getContextId() . '-' . $notification->getUserId() . '-' . $notification->getId();
+
+		return $uniqueKey;
+	}
+
+	/**
+	 * Creates and returns an encoded token that will be used to validate an unsubscribe url.
+	 * @param $notification Notification
+	 * @return string
+	 */
+	function createUnsubscribeToken($notification) {
+		$encodeString = $this->createUnsubscribeUniqueKey($notification);
+
+		$secret = Config::getVar('security', 'signed_page_key_secret', '');
+		$jwt = '';
+		if ($secret !== '') {
```

Comment:
Are you sure we want to return a blank token if the secret is not set? I would think that would be an error condition.

Reply:
See the message right above - The unsubscribe url will result a 404 page result

Reply:
OK, thanks!